src/addrman.h

```diff
@@ -410,9 +413,20 @@ friend class CAddrManTest;
 
         OverrideStream<Stream> s(&s_, s_.GetType(), stream_version);
 
-        unsigned char nKeySize;
-        s >> nKeySize;
-        if (nKeySize != 32) throw std::ios_base::failure("Incorrect keysize in addrman deserialization");
+        //! The maximum format this software can Unserialize().
+        static constexpr Format maximum_supported_format = Format::V3_BIP155;
+
+        uint8_t compat;
+        s >> compat;
+        const uint8_t lowest_compatible = compat - m_incompatibility_base;
+        if (lowest_compatible > maximum_supported_format) {
+            throw std::ios_base::failure(
+                strprintf("Unsupported format of addrman database: %u. It is compatible with "
```

Comment:
Thanks for adding an actual versioning mechanism, at least next time we won't have to resort to a hack like this!